staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_json.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+// JSON represents any valid JSON value.
+// These types are supported: bool, int64, float64, string, []interface{}, map[string]interface{} and nil.
+//
+// +protobuf=true
+// +protobuf.options.marshal=false
+// +protobuf.as=ProtoJSON
+// +protobuf.options.(gogoproto.goproto_stringer)=false
+// +k8s:conversion-gen=false
+type JSON struct {
+	Object interface{} `json:"-" protobuf:"-"`
+}
+
+// OpenAPISchemaType is used by the kube-openapi generator when constructing
+// the OpenAPI spec of this type.
+//
+// See: https://github.com/kubernetes/kube-openapi/tree/master/pkg/generators
+func (_ JSON) OpenAPISchemaType() []string {
+	// TODO: return actual types when anyOf is supported
+	return nil
+}
+
+// OpenAPISchemaFormat is used by the kube-openapi generator when constructing
+// the OpenAPI spec of this type.
+func (_ JSON) OpenAPISchemaFormat() string { return "" }
+
+func (j *JSON) DeepCopy() *JSON {
+	if j == nil {
+		return nil
+	}
+	return &JSON{Object: runtime.DeepCopyJSONValue(j.Object)}
+}
+
+func (j *JSON) DeepCopyInto(target *JSON) {
+	if target == nil {
+		return
+	}
+	if j == nil {
+		target.Object = nil // shouldn't happen
+	}
+	target.Object = runtime.DeepCopyJSONValue(j.Object)
+}
+
+func (j JSON) MarshalJSON() ([]byte, error) {
+	return json.Marshal(j.Object)
+}
+
+func (j *JSON) UnmarshalJSON(data []byte) error {
+	return json.Unmarshal(data, &j.Object)
+}
```

Comment:
Switching from `Raw []byte` to marshal/unmarshal means decoding/re-encoding now normalizes ordering in this field, right? I can't immediately think of issues with that, but wanted to note it in case you can.